### ISSUE TITLE
DestinationWrapper + AnimatedDestinationScope

### ISSUE THREAD

**Issue by matejdro** [raamcosta/compose-destinations#707]
Any chance we could get a `AnimatedDestinationScope` on the `DestinationWrapper`? 

That way, wrapper can do animations, like shared element transitions.

**Comment by raamcosta** [comment]
It should already be the case.

`DestinationScope` is a supertype of `AnimatedDestinationScope `.

In most cases, it will be `AnimatedDestinationScope` you can try a safe cast. 

Please let me know :)

**Comment by matejdro** [comment]
But this is an implementation detail. Ideally we don't want to depend on those, as they can break at any point at runtime.

**Comment by raamcosta** [comment]
I don't think this is an implementation detail. Both interfaces are part of the public API of the library and one extends the other. I cannot change this detail in the library without breaking clients, so I will not.

**Comment by matejdro** [comment]
Both are public, but it is an implementation detail that passed interface could be `AnimatedDestinationScope`

**Comment by raamcosta** [comment]
But if one is a subtype of the other, doesn't it stand to reason that it could be? :)

**Comment by matejdro** [comment]
it could be, but it's not guaranteed to be. Even if it is today, it can change without warning and starts breaking in production.

**Comment by raamcosta** [comment]
It can't be guaranteed because if the destination is a dialog or a bottom sheet, then it won't be. But the relationship between the two scope types is also part of the library API. It can't change without possibly breaking clients, so it won't be changed, meaning it will also be the case that it could be but never guaranteed.

Do you have ideas for chants that would make you more comfortable in using it?

**Comment by matejdro** [comment]
Ah I did not consider dialog or a bottom sheet. Thanks for the explanation.
